pkgs/by-name/wi/within/package.nix

```
@@ -0,0 +1,28 @@
+{
+  fetchFromGitHub,
+  lib,
+  stdenv,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "within";
+  version = "1.1.4";
+
+  src = fetchFromGitHub {
+    owner = "sjmulder";
+    repo = "within";
+    tag = finalAttrs.version;
+    hash = "sha256-UyOgEe07K1LW5IbB7ngxelp+9Njq/NPPkWw3sxAQyVY=";
+  };
+
```

Considering that you construct `src` by passing `finalAttrs.version` to `fetchFromGitHub`, it would be great to add some version checks. This will mitigate the security concern in issue #310373.
Since `within` doesn't print version information, we could improvise our own check with `CHANGELOG.md`:

```suggestion

  changelogPath = "CHANGELOG.md";
  checkVersionFromChangeLog = ''
    read -r < "$changelogPath"
    if [[ "''${REPLY:0:''${#version}+1}" == "$version " ]]; then
      echo "Found $version at the beginning of changelog $changelogPath"
    else
      echo "ERROR: $name: Changelog $changelogPath does not begin with $version"
      exit 1
    fi
  '';
  # If within supports `--help` or `--version`, we could switch to `versionCheckHook` without changing the name of the switch.
  dontVersionCheck = lib.match "^[0-9]([.][0-9]+)+" finalAttrs.version == null;
  preBuildPhases = lib.optional (!finalAttrs.dontVersionCheck) "checkVersionFromChangeLog";

```